File: NavTheoryClass/NavTheoryClass/高级转场动画/NavAdvanceAnimaViewCtr.m
//
//  NavAdvanceAnimaViewCtr.m
//  NavTheoryClass
//
//  Created by sy on 2019/1/5.
//  Copyright © 2019年 sy. All rights reserved.
//

#import "NavAdvanceAnimaViewCtr.h"
#import "SecondViewCtr.h"
#import "AnimationNavigationCtr.h"
@interface NavAdvanceAnimaViewCtr ()

@end

@implementation NavAdvanceAnimaViewCtr

+ (AnimationNavigationCtr*)newNavViewCtr{
    
 
    return [[AnimationNavigationCtr alloc] initWithRootViewController:[NavAdvanceAnimaViewCtr new]];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"高级动画";
    self.view.backgroundColor = UIColor.redColor;

    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"返回" style:UIBarButtonItemStylePlain target:self action:@selector(backAction)];
    
}


- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    
    [self.navigationController pushViewController:[SecondViewCtr new] animated:YES];
    
}

- (void)backAction{
    
    [self dismissViewControllerAnimated:YES completion:nil];
}


@end

File: NavAnimation/NavAnimation/Animation/NavAdvanceAnimaViewCtr.swift
//
//  NavAdvanceAnimaViewCtr.swift
//  NavAnimation
//
//  Created by Casey on 11/01/2019.
//  Copyright © 2019 newplatform. All rights reserved.
//

import UIKit

class NavAdvanceAnimaViewCtr: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.title = "animation"
        
        self.view.backgroundColor = .red
        
        self.navigationItem.leftBarButtonItem = UIBarButtonItem.init(title: "back", style: .plain, target: self, action: #selector(backAction))
    }
    
    
    @objc func backAction() {
        self.dismiss(animated: true, completion: nil)
        
    }
    
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        
        
        self.navigationController?.pushViewController(SecondViewCtr(), animated: true)
        
    }

   
}
